src/OrleansTestKit/Storage/TestStorage.cs

```diff
@@ -8,36 +8,38 @@ internal class TestStorage<TState> :
         IStorageStats,
         IStorage<TState>
     {
-        public TestStorageStats Stats { get; }
-
-        public TState State { get; set; }
-
-        public string Etag => throw new System.NotImplementedException();
-
-        public virtual bool RecordExists => throw new NotImplementedException();
-
         public TestStorage()
         {
-            Stats = new TestStorageStats() {Reads = -1};
+            Stats = new TestStorageStats() { Reads = -1 };
             InitializeState();
         }
 
+        public string Etag => throw new System.NotImplementedException();
+
+        public virtual bool RecordExists { get; set; }
+
+        public TState State { get; set; }
+
+        public TestStorageStats Stats { get; }
+
         public Task ClearStateAsync()
         {
             InitializeState();
             Stats.Clears++;
+            RecordExists = false;
             return Task.CompletedTask;
         }
 
-        public Task WriteStateAsync()
+        public Task ReadStateAsync()
         {
-            Stats.Writes++;
+            Stats.Reads++;
+            RecordExists = true;
```

Comment:
Shouldn't `RecordExists` be set in `WriteStateAsync`?

Reply:
oops.. your right!

Reply:
Ive fixed it!

Reply:
I'm having a hard time finding a place to add assertions... It looks like the only thing in the tests to user the the `TestStorage<TState>` is `TestStorage<TState>`, which subclasses `Grain<GreetingArchiveGrainState>` to get its state so its not possible for me to add assertions there.  Is there anywhere else you can suggest I look?

Reply:
So, I added a unit test using the `StorageManager` to create an instance of `TestStorage` to check that the flags is doing what it should.

Reply:
I can't drop into an IDE right now... but it looks like `TestKitSilo.StorageManager.GetStorage<TState>()` will return the `IStorage<TState>` instance. The `IStorage<TState>` interface exposes the `RecordExists` property.

It would be great to get new test cases that show that the flag being set, reset, and even unchanged based on the `WriteStateAsync`, `ClearStateAsync`, and `ReadAsyncState` calls, respectively.